hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/AltibaseSqlAstTranslator.java

```diff
@@ -0,0 +1,242 @@
+		renderComparisonEmulateIntersect( lhs, operator, rhs );
+	}
+
+	@Override
+	public void visitOver(Over<?> over) {
+		final Expression expression = over.getExpression();
+		if ( expression instanceof FunctionExpression && "row_number".equals( ( (FunctionExpression) expression ).getFunctionName() ) ) {
+			if ( over.getPartitions().isEmpty() && over.getOrderList().isEmpty()
+					&& over.getStartKind() == FrameKind.UNBOUNDED_PRECEDING
+					&& over.getEndKind() == FrameKind.CURRENT_ROW
+					&& over.getExclusion() == FrameExclusion.NO_OTHERS ) {
+				// Altibase doesn't allow an empty over clause for the row_number() function,
+				append( "row_number() over(order by 1)" );
+				return;
+			}
+		}
+		super.visitOver( over );
+	}
+
+	@Override
+	public void visitQuerySpec(QuerySpec querySpec) {
+		if ( shouldEmulateFetchClause( querySpec ) ) {
+			// Altibase does not support row_with_ties
+			emulateFetchOffsetWithWindowFunctions( querySpec, true );
+		}
+		else {
+			super.visitQuerySpec( querySpec );
+		}
+	}
+
+	protected boolean shouldEmulateFetchClause(QueryPart queryPart) {
+		// Check if current query part is already row numbering to avoid infinite recursion
+		return useOffsetFetchClause( queryPart ) && getQueryPartForRowNumbering() != queryPart
+				&& getDialect().supportsWindowFunctions() && !isRowsOnlyFetchClauseType( queryPart );
+	}
+
+	protected void renderTableGroupJoin(TableGroupJoin tableGroupJoin, List<TableGroupJoin> tableGroupJoinCollector) {
+		// Use join instead because Altibase does not support cross apply
+		appendSql( WHITESPACE );
+		if ( tableGroupJoin.getJoinType() != SqlAstJoinType.CROSS ) {
+			// No support for cross joins, so we emulate it with an inner join and always true on condition
+			appendSql( tableGroupJoin.getJoinType().getText() );
+		}
+		appendSql( "join " );
+
+		final Predicate predicate;
+		if ( tableGroupJoin.getPredicate() == null ) {
+			predicate = new BooleanExpressionPredicate( new QueryLiteral<>( true, getBooleanType() ) );
+		}
+		else {
+			predicate = tableGroupJoin.getPredicate();
+		}
+		if ( predicate != null && !predicate.isEmpty() ) {
+			renderTableGroup( tableGroupJoin.getJoinedGroup(), predicate, tableGroupJoinCollector );
+		}
+		else {
+			renderTableGroup( tableGroupJoin.getJoinedGroup(), null, tableGroupJoinCollector );
+		}
+	}
+
+	@Override
+	protected void renderPartitionItem(Expression expression) {
+		if ( expression instanceof Literal ) {
+			appendSql( "'0' || '0'" );
+		}
+		else if ( expression instanceof Summarization ) {
+			// This could theoretically be emulated by rendering all grouping variations of the query and
+			// connect them via union all but that's probably pretty inefficient and would have to happen
+			// on the query spec level
+			throw new UnsupportedOperationException( "Summarization is not supported by DBMS!" );
+		}
+		else {
+			expression.accept( this );
+		}
+	}
+
+	protected void prepareLimitOffsetParameters() {
```

Comment:
Instead of doing this, I would suggest you to override `renderOffsetExpression()` instead:

```java
	@Override
	protected void renderOffsetExpression(Expression offsetExpression) {
		appendSql( "1+" );
		offsetExpression.accept( this );
	}
```

Reply:
I think your suggestion solves this offset problem once and for all. Thanks!